windows: add tty unicode support for input

retep998 opened the pull request:
Switches from using ReadConsoleA to ReadConsoleW.
Uses an auxiliary buffer to store the temporary utf-16 string.

saghul commented on src/win/tty.c:
```diff
@@ -303,6 +303,8 @@ static DWORD CALLBACK uv_tty_line_read_thread(void* data) {
   uv_tty_t* handle;
   uv_req_t* req;
   DWORD bytes, read_bytes;
+  WCHAR utf16[8192 / 3];
```
Since this 8192 si related to line 320, can you please add a define with it?

retep998 replied:
Added MAX_INPUT_BUFFER_LENGTH

retep998 committed:
windows: add tty unicode support for input

Switches from using ReadConsoleA to ReadConsoleW.
Uses an auxiliary buffer to store the temporary utf-16 string.

Signed-off-by: Peter Atashian <retep998@gmail.com>

saghul commented:
LGTM. Can either of you guys have a quick look? @piscisaureus @orangemocha. Thanks!

piscisaureus commented:
lgtm too, ship it

saghul commented:
Thanks a lot @retep998, landed in https://github.com/joyent/libuv/commit/68cd6d6a6ff7b4088b210498f81002080733fe5f
